src/main/java/de/tum/in/www1/artemis/service/ExerciseService.java

```diff
@@ -93,6 +97,115 @@ public ExerciseService(ExerciseRepository exerciseRepository, ExerciseUnitReposi
         this.exerciseUnitRepository = exerciseUnitRepository;
         this.programmingExerciseRepository = programmingExerciseRepository;
         this.quizExerciseRepository = quizExerciseRepository;
+        this.ltiOutcomeUrlRepository = ltiOutcomeUrlRepository;
+        this.studentParticipationRepository = studentParticipationRepository;
+    }
+
+    /**
+     * Gets the subset of given exercises that a user is allowed to see
+     *
+     * @param exercises exercises to filter
+     * @param user      user
+     * @return subset of the exercises that a user allowed to see
+     */
+    public Set<Exercise> getSubsetThatUserIsAllowedToSee(Set<Exercise> exercises, User user) {
+        if (exercises == null || user == null) {
+            throw new IllegalArgumentException();
+        }
+        if (exercises.isEmpty()) {
+            return exercises;
+        }
+        Set<Course> courses = exercises.stream().map(Exercise::getCourseViaExerciseGroupOrCourseMember).collect(Collectors.toSet());
+        if (courses.size() != 1) {
+            throw new IllegalArgumentException("All exercises must be from the same course!");
+        }
+        Course course = exercises.stream().findFirst().get().getCourseViaExerciseGroupOrCourseMember();
```

Comment:
Can you just do `courses.get(0)` since you know that there's only 1 course?

Reply:
Yes but I see no benefit

Reply:
I think the intention is clearer and the code is more concise.

Reply:
Well if you both agree I change it in 1fe39af8c0744d175b3c0a694f15d1fb814806e1